lib/mongoid_session_store/store.rb

```diff
@@ -9,7 +9,7 @@ class Store < ActionDispatch::Session::AbstractStore
   private
 
     def generate_sid
-      BSON::ObjectId.new.to_s
+      SecureRandom.hex(32)
```

Comment:
Hi All!

Not sure why I am receiving these, is there any way for me to be removed?

Thanks

On Fri, Jun 24, 2016 at 9:35 AM, Pavel <notifications@github.com> wrote:

> In lib/mongoid_session_store/store.rb
> <https://github.com/SolarCS/mongo_session_store/pull/6#discussion_r68407093>
> :
>
> > @@ -9,7 +9,7 @@ class Store < ActionDispatch::Session::AbstractStore
> >    private
> >
> >      def generate_sid
> > -      BSON::ObjectId.new.to_s
> > +      SecureRandom.hex(32)
>
> About brute-force. Just think about this:
>
> $ curl -o /dev/null -s -w {time_connect}:%{time_starttransfer}:%{time_total}\\n  https://evolve.cipherhealth.com/home
> {time_connect}:1,527:1,527
>
> Even with three servers with multithreading 1.5 seconds responce with be a
> nightmare for hacker. Especially if session dies after two days.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/SolarCS/mongo_session_store/pull/6/files/e03633243e28193609e7532b65d816ee43944ee6#r68407093>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AIGVVAvnigYu0oDSzkseaGccfTFpR76Zks5qO-s5gaJpZM4I9sS0>
> .
>



-- 

Matt Danilo

*Account Strategist*

mdanilo@cipherhealth.com | 646.568.6143

CipherHealth <http://www.cipherhealth.com/> | 555 8th Avenue, Suite 701,
New York, NY

*The information contained in this email message is intended only for the
use of the individual or the entity named above. If the reader of this
message is not the intended recipient, or the employee or agent responsible
for delivering it to the intended recipient, you are hereby notified that
any dissemination, distribution or copying of this communication is
strictly prohibited. If you have received this communication in error,
please immediately notify me by email (mdanilo@cipherhealth.com
<mdanilo@cipherhealth.com>), and destroy the original message. Thank you.*